src/Core/src/Platform/Android/MauiTextView.cs

```diff
@@ -16,6 +18,25 @@ internal override void OnLayoutFormatted(bool changed, int l, int t, int r, int
 		{
 			LayoutChanged?.Invoke(this, new LayoutChangedEventArgs(l, t, r, b));
 		}
+		protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
+		{
+			if (MeasureSpec.GetMode(widthMeasureSpec) == MeasureSpecMode.AtMost && Layout is not null)
+			{
+				int maxWidth = (int)Math.Ceiling(GetMaxLineWidth(Layout)) + CompoundPaddingLeft + CompoundPaddingRight;
+				widthMeasureSpec = MeasureSpec.MakeMeasureSpec(maxWidth, MeasureSpecMode.AtMost);
+			}
+			base.OnMeasure(widthMeasureSpec, heightMeasureSpec);
+		}
+
+		static float GetMaxLineWidth(Layout layout)
+		{
+			float maxWidth = 0;
```

Comment:
A commente here like: "Calculates the maximum width needed to display the content based on the widest line." could be nice.

Reply:
A comment has been added